### ISSUE TITLE
Character escaping and julia-repl code blocks

### ISSUE THREAD

**Issue by tlienart** [tlienart/Franklin.jl#475]
With prerender=true and `julia-repl` it looks like the character escaping is going bananas as kindly pointed out by @shashi 

![highlight](https://user-images.githubusercontent.com/10897531/80799524-5b8d2f00-8ba7-11ea-85e3-49213defa3b5.png)

**Comment by shashi** [comment]
Note that this is not `julia-repl` code block though, it's actually ` ```julia ` inside a docstring, that I included into Franklin by doing `Markdown.html(docstring)`. So I'm not sure if `julia-repl` is affected.

**Comment by tlienart** [comment]
ohhhhhhh that explains it. So `Markdown.html` is a bit dumb; you could try using `Franklin.fd2html(markdown_string)` instead as it's more careful with how it treats code blocks. 

I need to try this as well to see what's going on.

**Comment by shashi** [comment]
Amazing just what I was looking for! Can I use franklin macros inside my docstrings then??? :heart_eyes:

**Comment by tlienart** [comment]
hmm ok so yes but this is a bit unchartered territory so you might encounter issues (hopefully not too many). Basically the key is that `fd2html` itself has its own scope by default, this might interfere with the scope of whatever page you're on. You can disable that by doing `fd2html(string, internal=true)`, this then uses the **current** scope and so will have access to everything that's available on that page.

Btw if you do not do `internal=true` calling `fd2html` within a page can be dangerous because it can wipe out the current scope. This is part of a broader issue that I need to look into of making sure that scopes are preserved but it requires some thinking.

**Comment by shashi** [comment]
Alright! Thanks for explaining that!! Yeah doesn't sound like fun. I'm okay with forcing codeblocks in docstrings to be 100% self contained.

**Comment by tlienart** [comment]
Ok no sorry in your case the safe path **is to use the internal flag** if you intend to use `fd2html` within your page process.

**Comment by newptcai** [comment]
I found this function is not very practical though . This seems to make serve() hangs forever and stop responding to ctrl c. Also changing doc string some times requires restart Julia to see the changes.

**Comment by tlienart** [comment]
^ no `fd2html` is instantaneous (<50ms), but you have to be a bit careful with where the code  is executed... this is why these features are not advertised widely at this point. You can always open an issue with reproducing code for discussion.

**Comment by newptcai** [comment]
I understand that. But when there is sophisticated plotting Code in the docstring, the process seems never finish.

There is another drawback which is to get output you need to put \output in docstring. That is perhaps not desirable.

I would recommend to leave example code in the docstring unevaluated if you use Franklin. Otherwise it is perhaps better to use Documenter.jl.

**Comment by tlienart** [comment]
Yeah I mean Franklin was not designed to be a competitor to Documenter. It can actually do stuff (as per the SymbolicUtils folks) but it's not magic. 

That being said, you could make this `hfun_docstring` as sophisticated as you want and essentially reproduce the whole of Documenter (hell you could even call Documenter functions from within Franklin), all this is possible but it will require a fair bit of effort as it's not something Franklin is explicitly supporting at this point.
